### PR TITLE
Replace invalid symbols in the labels for metadata visualization

### LINKED ISSUE #1625: Unable to visualize metadata (`Error: bad label format` and `CalledProcessError`)
### Environment Details
* SDV version: 1.4.0 (latest)
* Python version: 3.10
* Operating System: Linux (Colab Notebook)

### Error Description
When attempting to visualize my metadata for a single table, I am encountering a `CalledProcessError` from the underlying graphing library (graphviz). It appears that there is something misconfigured in the label as the first line is `Error: bad label format`.

### Steps to reproduce
Download the [Income dataset from Kaggle](https://www.kaggle.com/datasets/mastmustu/income). You can use the `train.csv` function to auto detect metadata and attempt to visualize it.

```python
import pandas as pd
from sdv.metadata import SingleTableMetadata

real_data = pd.read_csv('train.csv')
real_data['income_>50K'] = real_data['income_>50K'].astype(str)

metadata = SingleTableMetadata()
metadata.detect_from_dataframe(real_data)

metadata.visualize()
```

Output:
```
Error: bad label format 
...
CalledProcessError: '[PosixPath('dot'), '-Kdot', '-Tsvg']' returned non-zero exit status 1. ...
```

Full stack trace is attached below.
[stack_trace.txt](https://github.com/sdv-dev/SDV/files/12863158/stack_trace.txt)

### PR DESCRIPTION
**Pull request by fealho**
CU-86ayfuk2q
Resolve #1625.

### CODE REVIEW

**Comment by amontanez24** [comment]
Task linked: [CU-86ayfuk2q SDV - Unable to visualize metadata (`Error: bad label format` and `CalledProcessError`) #1625](https://app.clickup.com/t/86ayfuk2q)

**Review by amontanez24 — COMMENTED** [review]
Let's add a test but otherwise looks good!

File: sdv/metadata/visualization.py [hunk 1]
```diff
@@ -105,10 +107,10 @@ def visualize_graph(nodes, edges, filepath=None):
     )
 
     for name, label in nodes.items():
-        digraph.node(name, label=graphviz.escape(label))
+        digraph.node(name, label=_replace_special_characters(label))
```
Comment (amontanez24) [inline sdv/metadata/visualization.py:112]: have you tried using https://graphviz.readthedocs.io/en/stable/api.html#graphviz.nohtml

Comment (fealho) [reply sdv/metadata/visualization.py:112]: Yes, it doesn't address the problem.

Comment (amontanez24) [reply sdv/metadata/visualization.py:112]: I see. It seems like it works if you replace '>' with '\\>'. Then in the output graph the label looks correct 
Screenshot 2023-11-08 at 9 51 51 PM

Comment (fealho) [reply sdv/metadata/visualization.py:112]: Just a note, this approach removes the backslash from the column name. So if a column has `\<` in it, the output will be `<`.

Comment (amontanez24) [reply sdv/metadata/visualization.py:112]: I think that's ok since prior to this change if they had '\\>' in a label, it would only show '>' anyway.

**Review by amontanez24 — COMMENTED** [review]
Can we add an integration test for this case

File: tests/integration/metadata/test_visualization.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    # Run
+    metadata.visualize()
+
+
+def test_visualize_graph_for_multi_table():
+    """Test it runs when a column name contains `>`."""
+    # Setup
+    data1 = pd.DataFrame({'>': ['a', 'b', 'c']})
+    data2 = pd.DataFrame({'>': ['a', 'b', 'c']})
```
Comment (amontanez24) [inline tests/integration/metadata/test_visualization.py:21]: Can we test both symbols? and maybe the symbols mixed with text

**Review by amontanez24 — APPROVED** [review]

File: tests/integration/metadata/test_visualization.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    # Run
+    metadata.visualize()
+    metadata.validate()
+    model.fit(data)
+    model.sample(10)
+
+
+def test_visualize_graph_for_multi_table():
+    """Test it runs when a column name contains symbols."""
```
Comment (R-Palazzo) [inline tests/integration/metadata/test_visualization.py:24]: You can put a different message that the test before to be more specific

**Review by R-Palazzo — APPROVED** [review]
LGTM!

**Pull request merged**